test/spec/Feature/Auth/AudienceJwtSecretSpec.hs

```diff
@@ -3,43 +3,242 @@ module Feature.Auth.AudienceJwtSecretSpec where
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["youraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
+
+    it "succeeds when the audience claim has more than 1 element and one matches" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["notyouraudience", "youraudience", "anotheraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
+
+    it "fails when the audience claim has 1 element and it doesn't match" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["notyouraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith`
+          [json|{"code":"PGRST303","details":null,"hint":null,"message":"JWT not in audience"}|]
+          { matchStatus = 401 }
+
+
+    it "fails when the audience claim has more than 1 element and none matches" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["notyouraudience", "stillnotyouraudience", "anotheraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith`
+          [json|{"code":"PGRST303","details":null,"hint":null,"message":"JWT not in audience"}|]
+          { matchStatus = 401 }
+
+    it "fails when the audience claim is empty" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": []
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith`
+          [json|{"code":"PGRST303","details":null,"hint":null,"message":"JWT not in audience"}|]
+          { matchStatus = 401 }
+
+  context "when the audience has an invalid format"$ do
+    it "ignores the audience claim and succeeds" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": { "youraudience": true }
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
```

Comment:
This passes here, but it returns an error in v12:

```json
{
  "code":"PGRST301",
  "details":null,
  "hint":null,
  "message":"JWTClaimsSetDecodeError \"Error in $.aud: parsing StringOrURI failed, expected String, but encountered Object\""
}
```

Not sure if we should go back to returning an error or keep ignoring the `aud`. The latter needs documentation.

Reply:
Does the RFC say anything about an object?

I think we shoud fail and maintain the v12 behavior.

Reply:
>  I agree, we should return an error.

This is also valid for other registered claims, so I added the error handling for those too in a different commit.